test/jdk/java/awt/event/MouseWheelEvent/HWWheelScroll.java

```diff
@@ -0,0 +1,160 @@
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+/*
+ * @test
+ * @bug 4425654
+ * @summary Test wheel scrolling of heavyweight components
+ * @library /java/awt/regtesthelpers
+ * @build PassFailJFrame
+ * @run main/manual HWWheelScroll.html
+ */
+
+import java.awt.Choice;
+import java.awt.FileDialog;
+import java.awt.Frame;
+import java.awt.List;
+import java.awt.TextArea;
+import java.awt.Window;
+import java.lang.reflect.InvocationTargetException;
+import java.util.ArrayList;
+
+public class HWWheelScroll {
+    public static final int TEXT_TALL = 0;
+    public static final int TEXT_WIDE = 1;
+    public static final int TEXT_SMALL = 2;
+    public static final int TEXT_BIG = 3;
+    static String INSTRUCTIONS = """
+            Test for mouse wheel scrolling of heavyweight components with built-in
+            scrollbars or similar functionality that is controlled by guestures
+            such as Apple Magic Mouse or trackpad scrolling guesture.
+            Several windows containing either a TextArea, List, Choice, or a
+            FileDialog will appear. For each window, use the mouse wheel to
+            scroll its content, and then minimize it or move away
+            and continue with the next window.
+            Do not close any of the opened windows except the FileDialog.
+            For the FileDialog, first change to a directory with enough items that a
+            scrollbar appears.
+            Some of the other windows don't have enough text to warrant scrollbars,
+            but should be tested anyway to make sure no crash or hang occurs.
+            If all scrollbars scroll correctly, press "Pass", otherwise press "Fail".
+            """;
+
+    public static ArrayList<Window> initUI() {
+        ArrayList<Window> retValue = new ArrayList<>();
+        retValue.add(makeTextFrame(TextArea.SCROLLBARS_BOTH, TEXT_BIG));
+        retValue.add(makeTextFrame(TextArea.SCROLLBARS_BOTH, TEXT_TALL));
+        retValue.add(makeTextFrame(TextArea.SCROLLBARS_BOTH, TEXT_SMALL));
+        retValue.add(makeTextFrame(TextArea.SCROLLBARS_BOTH, TEXT_WIDE));
+        retValue.add(makeTextFrame(TextArea.SCROLLBARS_VERTICAL_ONLY, TEXT_TALL));
+        retValue.add(makeTextFrame(TextArea.SCROLLBARS_VERTICAL_ONLY, TEXT_SMALL));
+        retValue.add(makeTextFrame(TextArea.SCROLLBARS_HORIZONTAL_ONLY, TEXT_SMALL));
+        retValue.add(makeTextFrame(TextArea.SCROLLBARS_HORIZONTAL_ONLY, TEXT_WIDE));
+        retValue.add(makeListFrame(TEXT_TALL));
+        retValue.add(makeListFrame(TEXT_WIDE));
+        retValue.add(makeListFrame(TEXT_SMALL));
```

Comment:
I see these frames being generated and displayed, but I have a single empty & untitled frame. Which one is this supposed to be?
<img width="592" alt="Screenshot 2025-04-18 at 10 32 55 AM" src="https://github.com/user-attachments/assets/36058ddc-4ac3-4ae3-ab51-e52e8962ba07" />

Reply:
It's the parent frame for the FileDialog - you can try the scrolling inside or ignore it - it does not matter. Just do not close it until the test scenario is complete.

Reply:
Perhaps you should give it a title - see line 76. eg 
Frame f = new Frame("File Dialog Owner");

Reply:
Sure.